### ISSUE TITLE
False positive lint errors for rule RF02 in ORDER BY clause - Snowflake and PostgreSQL dialects

### ISSUE THREAD

**Issue by anthian** [sqlfluff/sqlfluff#6943]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

The rule RF02 (references.qualification) rises Lint errors for unqualified columns in ORDER BY clause, even if the column is a calculated column with aliases. This is not correct, almost for Postgree and Snowfalke SQL. 

What I do is downgrade this rule to a warning, because is correct in all other clauses but the ORDER BY one.

![Image](https://github.com/user-attachments/assets/aa9c7742-6a0e-4806-9279-49e85d9e6ce6)

![Image](https://github.com/user-attachments/assets/c76d83ce-6591-42a7-b46f-c9793235ab8c)

![Image](https://github.com/user-attachments/assets/404d1fdd-2897-4972-a345-16fe13883f8f)

### Expected Behaviour

I'm pretty sure that this occourse in other dialect, but I can't confirm that this behaviour is common to all dialects. The expected behaviour almost in the provided dialects should be not to check this rule on ORDER BY statement. 

### Observed Behaviour

The linter marks the unqualified columns in order by clause as errors. 

### How to reproduce

- Configure .sqlfluff file to activate RF02 rule
- Create a new SQL File with a multi-table statement SELECT-FROM-ORDER BY with table aliases. It could be a single table statement but it could interfer with RF03 rule. A simple stament could be: 

`SELECT T.COLUMN_NAME
FROM
	TEST_TABLE AS T
	INNER JOIN TEST_TABLE AS T2 ON (T.COLUMN_NAME = T2.COLUMN_NAME)
ORDER BY COLUMN_NAME;`

- Perform a Sqlfluff lint check for dialect "snowflake" or "postgresql"

### Dialect

- Snowlake
- PostgreSQL

### Version

3.3.1

### Configuration

`[sqlfluff]

# Supported dialects https://docs.sqlfluff.com/en/stable/perma/dialects.html
# Or run 'sqlfluff dialects'
dialect = snowflake

# One of [raw|jinja|python|placeholder]
templater = jinja

# Comma separated list of rules to exclude, or None
# See https://docs.sqlfluff.com/en/stable/perma/rule_disabling.html
# AM04 (ambiguous.column_count) and ST06 (structure.column_order) are
# two of the more controversial rules included to illustrate usage.
warnings = references.qualification
exclude_rules = ambiguous.column_count, structure.column_order, layout.select_targets, layout.cte_newline

# The standard max_line_length is 80 in line with the convention of
# other tools and several style guides. Many projects however prefer
# something a little longer.
# Set to zero or negative to disable checks.
max_line_length = 0

# CPU processes to use while linting.
# The default is "single threaded" to allow easy debugging, but this
# is often undesirable at scale.
# If positive, just implies number of processes.
# If negative or zero, implies number_of_cpus - specified_number.
# e.g. -1 means use all processors but one. 0 means all cpus.
processes = -1

# Disable large file skip byte limit
large_file_skip_byte_limit = 0

# If using the dbt templater, we recommend setting the project dir.
[sqlfluff:templater:dbt]
project_dir = ./

[sqlfluff:indentation]
# While implicit indents are not enabled by default. Many of the
# SQLFluff maintainers do use them in their projects.
indent_unit = tab
indented_joins = True
indented_ctes = True
indented_using_on = False
indented_on_contents = False
indented_then = False
indented_then_contents = False
allow_implicit_indents = True
trailing_comments = before

# Layout configuration
# See https://docs.sqlfluff.com/en/stable/layout.html#configuring-layout-and-spacing

[sqlfluff:layout:type:comma]
spacing_before = touch
line_position = trailing

[sqlfluff:layout:type:where_clause]
keyword_line_position = alone

[sqlfluff:layout:type:from_clause]
keyword_line_position = alone

[sqlfluff:layout:type:select_clause]
keyword_line_position = alone

[sqlfluff:layout:type:partitionby_clause]
keyword_line_position = leading

[sqlfluff:rules:aliasing.length]
min_alias_length = None
max_alias_length = None

# The default configuration for capitalisation rules is "consistent"
# which will auto-detect the setting from the rest of the file. This
# is less desirable in a new project and you may find this (slightly
# more strict) setting more useful.
# Typically we find users rely on syntax highlighting rather than
# capitalisation to distinguish between keywords and identifiers.
# Clearly, if your organisation has already settled on uppercase
# formatting for any of these syntax elements then set them to "upper".
# See https://stackoverflow.com/questions/608196/why-should-i-capitalize-my-sql-keywords-is-there-a-good-reason
[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = upper
[sqlfluff:rules:capitalisation.identifiers]
extended_capitalisation_policy = upper
[sqlfluff:rules:capitalisation.functions]
extended_capitalisation_policy = upper
[sqlfluff:rules:capitalisation.literals]
capitalisation_policy = upper
[sqlfluff:rules:capitalisation.types]
extended_capitalisation_policy = upper

# Reference qualification tweaks to avoid false positives in order by clauses
# See https://docs.sqlfluff.com/en/stable/reference/rules.html#rule-references.qualification
[sqlfluff:rules:references.consistent]
single_table_references = consistent
force_enable = false

# The default configuration for the not equal convention rule is "consistent"
# which will auto-detect the setting from the rest of the file. This
# is less desirable in a new project and you may find this (slightly
# more strict) setting more useful.
[sqlfluff:rules:convention.not_equal]
# Default to preferring the "ansi" (i.e. `<>`)
preferred_not_equal_style = ansi
`

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by anthian** [comment]
I had problems to add .sqlfluff file so I added as code, but github took it as markdownd text, so, sorry for the mess. 

** EDITED ** I have just deleted all comments to make it more readable.

**Comment by steveh-101** [comment]
Is this not correct? Because if you joined to another table with the same column name, which one would know to pick?

e.g. 

```sql
SELECT T.COLUMN_NAME, T2.COLUMN_NAME
FROM
	TEST_TABLE AS T
	INNER JOIN TEST_TABLE AS T2 ON (T.COLUMN_NAME = T2.COLUMN_NAME)
ORDER BY COLUMN_NAME;

**Comment by WittierDinosaur** [comment]
This is expected behaviour, the query you gave is ambiguous on the column name

**Comment by josepxhj** [comment]
Sure, sorry about the handmade example, that obviously has an ambigous column problem. I tried to simplify the example as much as possible, because normally i get this with calculate columns, that always have aliases.

It also arrises the problem with this query, that is correct:

SELECT T.COLUMN_NAME AS COLUMN_NAME_1, T2.COLUMN_NAME_2 AS COLUMN_NAME_2
FROM
	TEST_TABLE AS T
	INNER JOIN TEST_TABLE AS T2 ON (T.COLUMN_NAME = T2.COLUMN_NAME)
ORDER BY COLUMN_NAME_1;

The key point here is that the order by only accept the column names of the select clause with the same alias you defined on that clause. Hope it helps and sorry for.the inconvenience.
